text/0096-contract-metadata.md

```diff
@@ -0,0 +1,213 @@
+# Casper On-Chain Contract Metadata Standard
+
+## Summary
+
+[summary]: #summary
+
+This standard proposes a unified way for Casper smart contracts to self-describe with on-chain metadata, improving contract discoverability and trust. It defines optional metadata fields – `name`, `description`, `icon_uri`, and `project_uri` – which a contract can expose via dedicated entry points. These fields allow a contract to broadcast its human-readable name and purpose, along with links to an icon image and project website.
+
+All metadata is **immutable** (set once at deployment and never altered), ensuring consistency and integrity.
+
+By adopting this standard, Casper developers enable wallets, network explorers, and indexers to fetch and display contract information directly from the blockchain, much like how token standards (e.g. CEP-18 for fungible tokens or CEP-95 for NFTs) provide on-chain names and symbols, underscoring the community’s move toward rich on-chain identification.
+
+**Metadata Fields**:
+
+- `name`: A human-readable name for the contract (e.g. "Casper DEX").
```

Comment:
I'd prefix all the names with `contract_`, so it doesn't mix with entrypoints from the token standards.